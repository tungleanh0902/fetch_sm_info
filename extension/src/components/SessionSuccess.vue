<template>
  <div class="session session-success">
    <h2>Transaction Complete</h2>
    <div>
      <p>ðŸŽ‰ Your transaction was successfully signed!</p>
    </div>
    <div class="session-footer">
      <router-link to="/accounts">Back to my account list</router-link>
    </div>
  </div>
</template>

<script>
export default {
  name: `session-success`
}
</script>

<style scoped>
.session-success {
  padding: 2rem;
  background: var(--app-bg);
  color: var(--txt);
}

.session-success h2 {
  color: var(--bright);
  font-size: var(--h1);
  font-weight: 500;
  line-height: 3rem;
}
</style>
