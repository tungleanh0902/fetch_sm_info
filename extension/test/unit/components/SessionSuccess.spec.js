import { createLocalVue, shallowMount, RouterLinkStub } from '@vue/test-utils'
import Vuelidate from 'vuelidate'
import SessionSuccess from 'components/SessionSuccess'

describe(`SessionSuccess`, () => {
  const localVue = createLocalVue()
  localVue.use(Vuelidate)
  localVue.directive('focus', () => {})

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(SessionSuccess, {
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
  })

  it(`renders correctly`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has link back to accounts', async () => {
    expect(wrapper.findComponent(RouterLinkStub).props().to).toBe('/accounts')
  })
})
