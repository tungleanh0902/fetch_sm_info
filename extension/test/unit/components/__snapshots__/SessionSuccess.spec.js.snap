// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionSuccess renders correctly 1`] = `
<div
  class="session session-success"
>
  <h2>
    Transaction Complete
  </h2>
   
  <div>
    <p>
      ðŸŽ‰ Your transaction was successfully signed!
    </p>
  </div>
   
  <div
    class="session-footer"
  >
    <a>
      Back to my account list
    </a>
  </div>
</div>
`;
